torchvision/datasets/kinetics.py

```diff
@@ -35,29 +48,92 @@ class Kinetics400(VisionDataset):
                 └── class2
                     ├── clipx.avi
                     └── ...
-
+            If the split is not defined, it is appended using the split argument.
+        n_classes (int): select between Kinetics-400, Kinetics-600, and Kinetics-700 
+        split (str): split of the dataset to consider; currently supports ["train", "val"]
+        frame_rate (float): If not None, interpolate different frame rate for each clip.
         frames_per_clip (int): number of frames in a clip
         step_between_clips (int): number of frames between each clip
+        annotation_path (str): path to official Kinetics annotation file.
         transform (callable, optional): A function/transform that  takes in a TxHxWxC video
             and returns a transformed version.
+        download (bool): Download the official version of the dataset to root folder.
+        num_workers (int): Use multiple workers for VideoClips creation
+        _num_download_workers (int): Use multiprocessing in order to speed up download.
 
     Returns:
         tuple: A 3-tuple with the following entries:
 
-            - video (Tensor[T, H, W, C]): the `T` video frames
+            - video (Tensor[T, H, W, C]): the `T` video frames in torch.uint8 tensor
             - audio(Tensor[K, L]): the audio frames, where `K` is the number of channels
-              and `L` is the number of points
+              and `L` is the number of points in torch.float tensor
             - label (int): class of the video clip
+    
+    Raises:
+        RuntimeError: If ``download is True`` and the image archive is already extracted.
     """
 
-    def __init__(self, root, frames_per_clip, step_between_clips=1, frame_rate=None,
-                 extensions=('avi',), transform=None, _precomputed_metadata=None,
-                 num_workers=1, _video_width=0, _video_height=0,
-                 _video_min_dimension=0, _audio_samples=0, _audio_channels=0):
-        super(Kinetics400, self).__init__(root)
+    _FILES = {
+        400: "https://s3.amazonaws.com/kinetics/400/{split}/k400_{split}_path.txt",
+        600: "https://s3.amazonaws.com/kinetics/600/{split}/k600_{split}_path.txt",
+        700: "https://s3.amazonaws.com/kinetics/700_2020/{split}/k700_2020_{split}_path.txt",
+    }
+    _ANNOTATION = {
+        400: "https://s3.amazonaws.com/kinetics/400/annotations/{split}.csv",
+        600: "https://s3.amazonaws.com/kinetics/600/annotations/{split}.txt",
+        700: "https://s3.amazonaws.com/kinetics/700_2020/annotations/{split}.csv",
+    }
+
+    def __init__(
+        self,
+        root: str,
+        n_classes: int = 400,
```

Comment:
Although naturally an `int` would be correct, this also implies that I could set an arbitrary number. Thus I would go with `"400"`, `"600"`, and `"700"`.

```suggestion
        num_classes: str = "400",
```